casanovo/denovo/dataloaders.py

```
@@ -342,6 +370,369 @@ def _make_dataset(
+                        f_to_mzrt_to_pep[part][key].append(
+                            (window_center, cur_rt, 1)
+                        )
+                    else:
+                        f_to_mzrt_to_pep[part][key] = [
+                            (window_center, cur_rt, 1)
+                        ]
+
+        if cycle_time is None:
+            raise ValueError(
+                f"{mzml_file} does not contain MS1 scans required for DIA extraction"
+            )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)